pandas/core/frame.py

```diff
@@ -3944,11 +3945,11 @@ def isetitem(self, loc, value) -> None:
                 )
 
             for i, idx in enumerate(loc):
-                arraylike = self._sanitize_column(value.iloc[:, i])
+                arraylike, _ = self._sanitize_column(value.iloc[:, i])
```

Comment:
abstraction-wise ive been thinking of refs as being handled at the Manager/Block level.  is this wrong?  is there a viable way to keep this at that level?

Reply:
Not if we don't want to move sanitise column to the same level. We have to keep track of the refs when we extract the array from values to set